src/components/Pagination.jsx

```diff
@@ -0,0 +1,62 @@
+import PaginationButton from "./PaginationButton";
+import arrowLeftActive from "../assets/icons/icon-arrow-left-active.png";
+import arrowLeftInactive from "../assets/icons/icon-arrow-left-inactive.png";
+import arrowRightActive from "../assets/icons/icon-arrow-right-active.png";
+import arrowRightInactive from "../assets/icons/icon-arrow-right-inactive.png";
+
+const displayConfig = {
+  mobile: { itemCount: 4 },
+  tablet: { itemCount: 6 },
+  pc: { itemCount: 10 },
+};
+
+const Pagination = ({ page: currentPage, setPage, totalCount, display }) => {
+  const currentDisplay = displayConfig[display];
+  const lastPage = Math.ceil(totalCount / currentDisplay["itemCount"]);
+  const startIndex = Math.floor((currentPage - 1) / 5) * 5 + 1;
+  const lastIndex = Math.min(startIndex + 4, lastPage);
+  const pageList = [];
+  for (let i = startIndex; i <= lastIndex; i++) {
+    pageList.push(i);
+  }
+  return (
+    <ul className="flex gap-4">
+      <li>
+        <PaginationButton
+          onClick={() => setPage(lastIndex - 5)}
+          isActive={currentPage > 5}
+        >
+          <img
+            src={currentPage <= 5 ? arrowLeftInactive : arrowLeftActive}
+            alt="왼쪽 화살표"
+            className="size-16"
+          />
+        </PaginationButton>
+      </li>
+      {pageList.map((page) => (
+        <li key={page}>
+          <PaginationButton
+            onClick={() => setPage(page)}
+            isCurrentPage={currentPage === page}
+            isActive={true}
+          >
+            {page}
+          </PaginationButton>
+        </li>
+      ))}
```

Comment:
💊 제안
정적 요소와 동적 요소가 혼합되어 작성되면, 추후 유지보수 시 어디서 무엇이 생성되는지 파악하기 어려울 수 있습니다.
가급적 전체 페이지 버튼들을 한 번에 생성하는 방식-Array.map, Array.from-으로 작성하시는 것을 추천드려요!

가장 추천드리는 방식은 아래처럼 jsx에서 바로 작성하시는 거에요~


```suggestion
  const pageList = Array.from({ length: lastIndex - startIndex}, (_, i) => startIndex + i );
  
  return (
    <ul className="flex gap-4">
      <li>
        <PaginationButton
          onClick={() => setPage(lastIndex - 5)}
          isActive={currentPage > 5}
        >
          <img
            src={currentPage <= 5 ? arrowLeftInactive : arrowLeftActive}
            alt="왼쪽 화살표"
            className="size-16"
          />
        </PaginationButton>
      </li>
      {pageList.map((page) => (
        <li key={page}>
          <PaginationButton
            onClick={() => setPage(page)}
            isCurrentPage={currentPage === page}
            isActive={true}
          >
            {page}
          </PaginationButton>
        </li>
      ))}
```